ocm-container.sh

```diff
@@ -234,22 +234,64 @@ then
   fi
 fi
 
-## Check for backplane config dir override and then mount the directory if it exists
+## Check for backplane config dir override
 if [ -z "$BACKPLANE_CONFIG_DIR" ]
 then
   BACKPLANE_CONFIG_DIR=$DEFAULT_BACKPLANE_CONFIG_DIR_LOCATION
 fi
-
-if [ -d "$BACKPLANE_CONFIG_DIR" ]
+## Set ocm url
+if [ -z $OCM_URL ]
 then
-  BACKPLANE_CONFIG_MOUNT="-v $BACKPLANE_CONFIG_DIR:/root/.config/backplane:ro"
-  if [ -z $OCM_URL ] || [ $OCM_URL == "production" ]
-  then
-    BACKPLANE_CONFIG_MOUNT="$BACKPLANE_CONFIG_MOUNT -e BACKPLANE_CONFIG=/root/.config/backplane/config.json"
-  else
-    BACKPLANE_CONFIG_MOUNT="$BACKPLANE_CONFIG_MOUNT -e BACKPLANE_CONFIG=/root/.config/backplane/config.$OCM_URL.json"
-  fi
+  OCM_URL="production"
+fi
+## Set the mount path
+BACKPLANE_CONFIG_MOUNT="-v $BACKPLANE_CONFIG_DIR:/root/.config/backplane:ro"
+## Set backplane config env var based on the environment
+if [ $OCM_URL == "production" ]
+then
+  BACKPLANE_CONFIG="config.json"
+else
+  BACKPLANE_CONFIG="config.$OCM_URL.json"
 fi
+## Create backplane config if missing
+if [ ! -f $BACKPLANE_CONFIG_DIR/$BACKPLANE_CONFIG ]; then
+    echo "Cannot find backplane config file at $BACKPLANE_CONFIG_DIR/$BACKPLANE_CONFIG";
+    echo "Setting up backplane config for ${OCM_URL}..."
+    read -t 300 -p 'backplane url: ' BACKPLANE_CONFIG_URL
+    if [[ $? -gt 128 ]]
+    then
+      echo -e "\nTimeout waiting for backplane url"
+      exit 1
+    fi
+    read -t 300 -p 'proxy-url: ' BACKPLANE_CONFIG_PROXY_URL
+    if [[ $? -gt 128 ]]
+    then
+      echo -e "\nTimeout waiting for backplane proxy url"
+      exit 1
+    fi
+    read -t 300 -p 'assume-initial-arn: ' BACKPLANE_CONFIG_ASSUME_ARN
+    if [[ $? -gt 128 ]]
+    then
+      echo -e "\nTimeout waiting for assume initial arn"
+      exit 1
+    fi
+    ## Ensure backplane conf dir exits
+    if [ ! -d "$BACKPLANE_CONFIG_DIR" ]
+    then
+      mkdir -p $BACKPLANE_CONFIG_DIR
+    fi
+    ## Create backplane configuration
+    cat << EOF > $BACKPLANE_CONFIG_DIR/$BACKPLANE_CONFIG
+{
+    "assume-initial-arn": "${BACKPLANE_CONFIG_ASSUME_ARN}",
+    "proxy-url": "${BACKPLANE_CONFIG_PROXY_URL}",
+    "session-dir": "~/.config/backplane/session",
+    "url": "${BACKPLANE_CONFIG_URL}"
```

Comment:
FYI : This setting will be deprecated in the next release. Anyway it's not harmful to set the URL but will show `warn` messages: [ref](https://issues.redhat.com/browse/OSD-19188)

Reply:
Thanks for the heads up, fixed in https://github.com/openshift/ocm-container/commit/c1335144a235de1fa7cea096c1b6c73ae6921f7f